beater/beater.go

```diff
@@ -217,6 +218,51 @@ func (bt *beater) registerPipelineCallback(b *beat.Beat) error {
 	return err
 }
 
+func (bt *beater) initTracing(b *beat.Beat) (*apm.Tracer, *tracerServer, error) {
+	var err error
+	tracer := b.Instrumentation.Tracer()
+	listener := b.Instrumentation.Listener()
+
+	if !tracer.Active() && bt.config != nil {
+		tracer, listener, err = initLegacyTracer(b.Info, bt.config)
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+
+	tracerServer := newTracerServer(bt.config, listener)
+	return tracer, tracerServer, nil
+}
+
+// initLegacyTracer exists for backwards compatibility and it should be removed in 8.0
+// it does not instrument the beat output
+func initLegacyTracer(info beat.Info, cfg *config.Config) (*apm.Tracer, net.Listener, error) {
+	selfInstrumentation := cfg.SelfInstrumentation
+	if selfInstrumentation == nil || !selfInstrumentation.IsEnabled() {
+		return apm.DefaultTracer, nil, nil
+	}
+	conf := common.MustNewConfigFrom(cfg.SelfInstrumentation)
```

Comment:
`MustNewConfigFrom` panics in case an error occurs. So far we use it in tests or when the whole input config has fixed values. I don't think we shoul be using it with user provided input and suggest to instead use `NewConfigFrom`  with the additional error check.